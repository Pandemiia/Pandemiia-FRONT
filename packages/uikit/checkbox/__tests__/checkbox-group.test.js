import React from 'react'
import { shallow } from 'enzyme'
import CheckboxGroup from '../checkbox-group'

describe('uikit/checkbox-group', () => {
  it('should renders correctly', () => {
    const options = [
      { value: 'val1', label: 'Value1' },
      { value: 'val2', label: 'Value2' }
    ]
    const wrapper = shallow(
      <CheckboxGroup
        disabled
        options={options}
        value={['val1', 'val2']}
        onChange={() => {}}
      />
    )

    wrapper
      .find('Checkbox')
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ value: [] })

    wrapper
      .find('Checkbox')
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })
  })
})
