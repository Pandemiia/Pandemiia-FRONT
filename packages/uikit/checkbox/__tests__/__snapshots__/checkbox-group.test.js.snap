// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/checkbox-group should renders correctly 1`] = `
<div
  className="checkboxGroup inline disabled"
  data-role="CheckboxGroup"
>
  <label
    className="checkboxWrapper"
    key="val1"
  >
    <Checkbox
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={true}
    />
    <Text
      align="left"
      className="label"
      component="p"
      intent="primary"
      size="m"
      typeface="base"
    >
      Value1
    </Text>
  </label>
  <label
    className="checkboxWrapper"
    key="val2"
  >
    <Checkbox
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={true}
    />
    <Text
      align="left"
      className="label"
      component="p"
      intent="primary"
      size="m"
      typeface="base"
    >
      Value2
    </Text>
  </label>
</div>
`;
